src/authentication/components/ExecutionConfigModal.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useForm } from "react-hook-form";
+import {
+  ActionGroup,
+  AlertVariant,
+  Button,
+  ButtonVariant,
+  Form,
+  FormGroup,
+  Modal,
+  ModalVariant,
+  TextInput,
+  ValidatedOptions,
+} from "@patternfly/react-core";
+import { CogIcon, TrashIcon } from "@patternfly/react-icons";
+
+import type AuthenticatorConfigRepresentation from "keycloak-admin/lib/defs/authenticatorConfigRepresentation";
+import type AuthenticatorConfigInfoRepresentation from "keycloak-admin/lib/defs/authenticatorConfigInfoRepresentation";
+import type { ConfigPropertyRepresentation } from "keycloak-admin/lib/defs/authenticatorConfigInfoRepresentation";
+import type { ExpandableExecution } from "../execution-model";
+import { useAdminClient, useFetch } from "../../context/auth/AdminClient";
+import { useAlerts } from "../../components/alert/Alerts";
+import { HelpItem } from "../../components/help-enabler/HelpItem";
+
+export const ExecutionConfigModal = ({
+  execution,
+}: {
+  execution: ExpandableExecution;
+}) => {
+  const { t } = useTranslation("authentication");
+  const adminClient = useAdminClient();
+  const { addAlert, addError } = useAlerts();
+
+  const [show, setShow] = useState(false);
+  const [config, setConfig] = useState<AuthenticatorConfigRepresentation>();
+  const [configDescription, setConfigDescription] =
+    useState<AuthenticatorConfigInfoRepresentation>();
+
+  const { register, errors, setValue, handleSubmit } = useForm();
+
+  const setupForm = (
+    configDescription: any,
+    config?: AuthenticatorConfigRepresentation
+  ) => {
+    configDescription.properties.map(
+      (property: ConfigPropertyRepresentation) => {
+        setValue(
+          property.name!,
+          config?.config?.[property.name!] || property.defaultValue || ""
```

Comment:
Note that this will use values that are empty strings from the config. If you want to explicitly check if it is a nullish value then the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) might be a better option.
```suggestion
          config?.config?.[property.name!] ?? property.defaultValue ?? ""
```

Reply:
do we really care about the difference between ?? and ||

Reply:
Depends if you want to allow empty strings here (if they are values that could occur).